tests/testlib/s2n_testlib.h

```diff
@@ -245,3 +245,40 @@ extern const s2n_parsed_extension EMPTY_PARSED_EXTENSIONS[S2N_PARSED_EXTENSIONS_
 int s2n_kem_recv_public_key_fuzz_test(const uint8_t *buf, size_t len, struct s2n_kem_params *kem_params);
 int s2n_kem_recv_ciphertext_fuzz_test(const uint8_t *buf, size_t len, struct s2n_kem_params *kem_params);
 int s2n_kem_recv_ciphertext_fuzz_test_init(const char *kat_file_path, struct s2n_kem_params *kem_params);
+
+/* kTLS */
+#define S2N_TEST_KTLS_MOCK_HEADER_SIZE        3
+#define S2N_TEST_KTLS_MOCK_HEADER_LENGTH_SIZE 2
+
+/* The record_type is communicated via ancillary data when using kTLS. For this
+ * reason s2n must use `send/recvmsg` syscalls rather than `send/read`. To mimic
+ * the send/recvmsg calls more accurately, we mock the socket via two separate
+ * buffers: data_buffer and ancillary_buffer.
+ *
+ * The mock implementation, uses 3 bytes with a tag + len format to represent
+ * each record. The first byte(tag) is the record_type and the next two represent
+ * the length of the record. Length is represented as a u16 to capture the max
+ * possible TLS record length.
+ *
+ * ancillary_buffer memory layout:
+ *    [      u8      |            u16            ]
+ *      record_type             length
+ */
+struct s2n_test_ktls_io_stuffer {
+    struct s2n_stuffer ancillary_buffer;
```

Comment:
I don't get why this is a buffer inside a struct. Couldn't you represent it as a uint8_t and a uint_16_t, like in the comment?

Reply:
There can be multiple records that we send/recv. So each record is `[u8|u16]`.

ancillary_buf with 3 records:
```
[ [record] [record] [record] ]
[ [u8|u16] [u8|u16] [u8|u16] ]
```

similarly the data buffer can have multiple records that are simply stored in a single array